sdk/go/core/docsgen/api/types/ydk_types.rst

```diff
@@ -268,44 +268,172 @@ These are how YANG types are represented in Go.
 
 .. function:: SetParent(entity, parent Entity)
 
     SetParent sets the given :ref:`Entity <types-entity>` parent field to the given parent :ref:`Entity <types-entity>`
 
-.. function:: HasDataOrFilter(entity Entity)
+    :param entity: An instance of :ref:`Entity <types-entity>`
 
-    HasDataOrFilter returns a bool representing whether the :ref:`Entity <types-entity>` or any of its children have their data/filter set
+.. function:: HasDataOrFilter(entity Entity) bool
 
-    :return: (``bool``) A Go boolean.
+    :param entity: An instance of :ref:`Entity <types-entity>`
+    :return: A Go boolean representing whether the :ref:`Entity <types-entity>` or any of its children have their data/filter set
 
 .. function:: GetEntityPath(entity Entity)
 
-    GetEntityPath returns an EntityPath struct for the given entity
+    :param entity: An instance of :ref:`Entity <types-entity>`
+    :return: :go:struct:`EntityPath` for the given entity
 
-    :return: :go:struct:`EntityPath`
+.. function:: GetChildByName(entity Entity, childYangName string, segmentPath string) Entity
 
-.. function:: GetChildByName(entity Entity, childYangName, segmentPath string)
+    Finds entity's child entity by name and segment path
 
-    GetChildByName takes an :ref:`Entity <types-entity>` and returns the child :ref:`Entity <types-entity>` described by the given childYangName and segmentPath or nil if there is no match
-
-    :return: :ref:`Entity <types-entity>`
+    :param entity: An instance of :ref:`Entity <types-entity>`
+    :param childYangName: The `YangName` of the child entity
+    :param segmentPath: The `SegmentPath` value of the child entity
+    :return: :ref:`Entity <types-entity>` described by the given `childYangName` and `segmentPath` or ``nil`` if there is no match
 
 .. function:: SetValue(entity Entity, valuePath string, value interface{})
 
-    Given an entity, SetValue sets the leaf specified by valuePath to the given value
+    Sets leaf value
+
+    :param entity: An instance of :ref:`Entity <types-entity>`
+    :param valuePath: The :go:struct:`YLeaf` `name` value
+    :param value: Instance of value interface
+
+.. function:: EntityToString(entity Entity) string
+
+    Utility function to get string representation of the entity.
+
+    :return:  Go ``string`` in format: "Type: `entity-instance-type`, Path: `entity-segment-path`".
+
+
+
+.. _entity-collection:
+
+.. go:struct:: EntityCollection
+
+    Type `EntityCollection` along with its methods implements ordered map collection of entities. The string value of entity `SegmentPath` serves as a map key for the entity. Ordered means, the collection retains order of entities, in which they were added to collection. 
+    
+    The `EntityCollection` type has two aliases - `Config` and `Filter`.
+
+    .. function:: GetEntityData() *CommonEntityData
+
+        Implements Entity interface.
```

Comment:
I think it's better to include the link to to the Entity interface:

Implements :ref:\`Entity <types-entity>\` interface

Reply:
Fixed.

Reply:
I did not see the change in your latest push - bumping this comment so it doesn't get buried

Reply:
It is in my fork: https://github.com/ygorelik/ydk-gen/commit/d227bd9997518478a9f1055198c7f38cc07b5fc6